src/components/data/RenameDialog.js

```diff
@@ -0,0 +1,163 @@
+    const classes = useStyles();
+    const baseId = ids.RENAME_DLG;
+    const { t } = useTranslation("data");
+
+    const [resourceRename, { status: renameStatus }] = useMutation(rename, {
+        onSuccess: (data, { resetForm }) => {
+            setRenameError(null);
+            announce({
+                text: t("asyncRenamePending"),
+                variant: AnnouncerConstants.INFO,
+            });
+            resetForm();
+            onRenamed();
+        },
+        onError: (error) => {
+            const text =
+                getErrorCode(error) === ERROR_CODES.ERR_EXISTS
+                    ? t("resourceExists", { path: getParentPath(path) })
+                    : t("renameFail");
+            setRenameError(text);
+        },
+    });
+
+    const handleRename = ({ name }, { resetForm }) => {
+        const newPath = `${getParentPath(path)}/${name}`;
+        resourceRename({ dest: newPath, source: path, resetForm });
+    };
+
+    const validate = ({ name }) => {
+        const validationError = validateDiskResourceName(name, t);
+        return validationError ? { name: validationError } : {};
+    };
+
+    const isLoading = isQueryLoading(renameStatus);
+    const textFieldProps = {};
+    if (renameError) {
+        textFieldProps.error = true;
+        textFieldProps.helperText = renameError;
+    }
+
+    return (
+        <Formik
+            initialValues={{ name: parseNameFromPath(path) }}
+            validate={validate}
+            onSubmit={handleRename}
+            enableReinitialize
+        >
+            {({ handleSubmit }) => {
+                return (
+                    <Form>
+                        <DEDialog
+                            open={open}
+                            onClose={onClose}
+                            maxWidth="sm"
+                            fullWidth
+                            title={t("rename")}
+                            actions={
+                                <>
+                                    <Button
+                                        id={build(baseId, ids.CANCEL_BTN)}
+                                        onClick={onClose}
+                                    >
+                                        {t("cancel")}
+                                    </Button>
+                                    <Button
+                                        id={build(baseId, ids.RENAME_BTN)}
+                                        color="primary"
+                                        type="submit"
+                                        onClick={handleSubmit}
+                                    >
+                                        {t("rename")}
+                                    </Button>
+                                </>
+                            }
+                        >
+                            <Typography
+                                classes={{ root: classes.bottomPadding }}
```

Comment:
Is this different from setting `gutterBottom={true}`?
https://material-ui.com/api/typography/#props

Reply:
Hmm entirely sure but `gutterBottom` is for margins and this classes is setting `padding`. I copy-pasted from `CreateFolderDialog`.

Reply:
👌 Well I just messed around with it in Storybook real quick, and they seem to have the same effect to me 🤷

Reply:
May be we can get rid of `custom` style and replace it `gutterBottom` it in another PR from these 2 dialog 👍🏽

Reply:
Ah neat!  I either didn't know or forgot about `gutterBottom` 😄 I'll keep it in mind going forward